llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -2985,6 +2985,13 @@ Instruction *InstCombinerImpl::foldSelectOfBools(SelectInst &SI) {
       return BinaryOperator::CreateOr(CondVal, FalseVal);
     }
 
+    if (match(CondVal, m_OneUse(m_Select(m_Value(A), m_One(), m_Value(B)))) &&
+        impliesPoison(FalseVal, B)) {
```

Comment:
Can you check the compile time impact?

Reply:
@nikic could you test the compile time impact of this patch?

Reply:
Sorry for the delay, here are the results: https://llvm-compile-time-tracker.com/compare.php?from=03ffb82c9e0d363c97ca37ede46719236616c88e&to=79752c9ee833172600523f038a6f83d64f657f9d&stat=instructions:u